ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -68,17 +69,19 @@ public AggregateAnalysisResult analyze(
     private final MutableAggregateAnalysis aggregateAnalysis = new MutableAggregateAnalysis();
     private final FunctionRegistry functionRegistry;
     private final Set<Expression> groupBy;
+    private boolean foundExpressionInGroupBy = false;
```

Comment:
why can't this be part of the `Visitor` directly? it seems weird for this to carry over between process steps (lines 95 -> 109)

Reply:
+1